File: back_up/filter/filter-names.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"time"
)

func filter(vs []string, f func(string) bool) []string {
	vsf := vs[:0]
	for _, v := range vs {
		if f(v) {
			vsf = append(vsf, v)
		}
	}
	return vsf
}

func main() {
	file, err := os.Open("../data/names.txt")
	if err != nil {
		panic("error in opening the file")
	}

	names := []string{}
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		names = append(names, scanner.Text())
	}
	originalCount := len(names)

	start := time.Now()

	names = filter(names, func(name string) bool { return !strings.Contains(name, "A") })

	elapsed := time.Since(start)
	fmt.Println("original count is", originalCount)
	fmt.Println("filtered count is", len(names))
	fmt.Println(elapsed)
}

File: back_up/filter/FilterNames.cs
using System;
using System.IO;
using System.Linq;
using System.Diagnostics;

namespace filter
{
    class Program
    {
        static void Main(string[] args)
        {
            try 
            {
                var fileName = "../data/names.txt";
                var line = File.ReadAllLines(fileName);
                var originalSize = line.Length;

                Stopwatch sw = new Stopwatch();
                sw.Start();

                line = line.Where(l => !l.Contains("A")).ToArray();

                sw.Stop();
                Console.WriteLine("Elapsed={0} ms",sw.Elapsed.Milliseconds);

                Console.WriteLine("Original Size : " + originalSize);
                Console.WriteLine("Filtered Size : " + line.Length);

            }
            catch(Exception e)
            {
                Console.WriteLine(e.ToString());
            }
        }
    }
}
